kong/plugins/jwt/jwt_parser.lua

```diff
@@ -119,7 +152,40 @@ local alg_verify = {
     local pkey, _ = openssl_pkey.new(key)
     assert(#signature == 96, "Signature must be 96 bytes.")
     return pkey:verify(signature, data, "sha384", nil, { ecdsa_use_raw = true })
-  end
+  end,
+
+  ES512 = function(data, signature, key)
+    --  Signing and validation with the ECDSA P-384 SHA-384 and ECDSA P-521
+    --  SHA-512 algorithms is performed identically to the procedure for
+    --  ECDSA P-256 SHA-256 -- just using the corresponding hash algorithms
+    --  with correspondingly larger result values.  For ECDSA P-384 SHA-384,
+    --  R and S will be 384 bits each, resulting in a 96-octet sequence.  For
+    --  ECDSA P-521 SHA-512, R and S will be 521 bits each, resulting in a
+    --  132-octet sequence.
+    local pkey, _ = openssl_pkey.new(key)
+    assert(#signature == 132, "Signature must be 132 bytes.")
+    return pkey:verify(signature, data, "sha512", nil, { ecdsa_use_raw = true })
+  end,
+
+  PS256 = function(data, signature, key)
+    local pkey, _ = openssl_pkey.new(key)
+    assert(pkey, "Consumer Public Key is Invalid")
+    assert(#signature == 256, "Signature must be 256 bytes")
+    return pkey:verify(signature, data, "sha256", openssl_pkey.PADDINGS.RSA_PKCS1_PSS_PADDING)
+  end,
+  PS384 = function(data, signature, key)
+    local pkey, _ = openssl_pkey.new(key)
+    assert(pkey, "Consumer Public Key is Invalid")
+    assert(#signature == 256, "Signature must be 256 bytes")
+    return pkey:verify(signature, data, "sha384", openssl_pkey.PADDINGS.RSA_PKCS1_PSS_PADDING)
+  end,
+  PS512 = function(data, signature, key)
+    local pkey, _ = openssl_pkey.new(key)
+    assert(pkey, "Consumer Public Key is Invalid")
+    assert(#signature == 256, "Signature must be 256 bytes")
+    return pkey:verify(signature, data, "sha512", openssl_pkey.PADDINGS.RSA_PKCS1_PSS_PADDING)
+  end,
```

Comment:
it might be helpful to include a comment that justifies the 256 bytes signature size validation like we have for ES above

Reply:
The signature length assertion is actually incorrect here. We don't restrict users to certain key-sizes which result in different signature lengths. Removed it 👍🏼